core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -0,0 +1,719 @@
+                    return new RLMTaskWithFuture(task, future);
+                }
+        );
+        convertToLeaderOrFollower.accept(rlmTaskWithFuture.rlmTask);
+    }
+
+    static class RLMTaskWithFuture {
+
+        private final RLMTask rlmTask;
+        private final Future<?> future;
+
+        RLMTaskWithFuture(RLMTask rlmTask, Future<?> future) {
+            this.rlmTask = rlmTask;
+            this.future = future;
+        }
+
+        public void cancel() {
+            rlmTask.cancel();
+            try {
+                future.cancel(true);
+            } catch (Exception ex) {
+                LOGGER.error("Error occurred while canceling the task: {}", rlmTask, ex);
+            }
+        }
+
+    }
+
+    /**
+     * Closes and releases all the resources like RemoterStorageManager and RemoteLogMetadataManager.
+     */
+    public void close() {
+        synchronized (this) {
+            if (!closed) {
+                leaderOrFollowerTasks.values().forEach(RLMTaskWithFuture::cancel);
+                Utils.closeQuietly(remoteLogStorageManager, "RemoteLogStorageManager");
+                Utils.closeQuietly(remoteLogMetadataManager, "RemoteLogMetadataManager");
+                Utils.closeQuietly(indexCache, "RemoteIndexCache");
+                try {
+                    rlmScheduledThreadPool.shutdown();
+                } catch (InterruptedException e) {
+                    // ignore
+                }
+                leaderOrFollowerTasks.clear();
+                closed = true;
+            }
+        }
+    }
+
+    static class RLMScheduledThreadPool {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(RLMScheduledThreadPool.class);
+        private final int poolSize;
+        private final ScheduledThreadPoolExecutor scheduledThreadPool;
+
+        public RLMScheduledThreadPool(int poolSize) {
+            this.poolSize = poolSize;
+            scheduledThreadPool = createPool();
+        }
+
+        private ScheduledThreadPoolExecutor createPool() {
+            ScheduledThreadPoolExecutor threadPool = new ScheduledThreadPoolExecutor(poolSize);
+            threadPool.setRemoveOnCancelPolicy(true);
+            threadPool.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
+            threadPool.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
+            threadPool.setThreadFactory(new ThreadFactory() {
+                private final AtomicInteger sequence = new AtomicInteger();
+
+                public Thread newThread(Runnable r) {
+                    return KafkaThread.daemon("kafka-rlm-thread-pool-" + sequence.incrementAndGet(), r);
+                }
+            });
+
+            return threadPool;
+        }
+
+        public ScheduledFuture scheduleWithFixedDelay(Runnable runnable, long initialDelay, long delay, TimeUnit timeUnit) {
```

Comment:
I know I'm late to the party, but I can see that `ScheduledFuture` is parametrized, but used "raw" in here.
I think this should be like this instead:
> public ScheduledFuture<?> scheduleWithFixedDelay(Runnable runnable, long initialDelay, long delay, TimeUnit timeUnit) {

I'll submit a fix for this